sharding/notary/notary.go

```diff
@@ -151,7 +205,287 @@ func joinNotaryPool(client mainchain.Client) error {
 	if err != nil {
 		return fmt.Errorf("unable to deposit eth and become a notary: %v", err)
 	}
+	receipt, err := client.TransactionReceipt(tx.Hash())
+	if err != nil {
+		return err
+	}
+	if receipt.Status == uint(0) {
+
+		return errors.New("Transaction was not successful, unable to deposit ETH and become a notary")
+
+	}
+
+	if inPool, err := isAccountInNotaryPool(client); !inPool || err != nil {
+		if !inPool {
+
+			return errors.New("Account has not been able to be deposited in Notary Pool")
+		}
+
+		return err
+	}
+
 	log.Info(fmt.Sprintf("Deposited %dETH into contract with transaction hash: %s", new(big.Int).Div(sharding.NotaryDeposit, big.NewInt(params.Ether)), tx.Hash().String()))
 
 	return nil
 }
+
+// leaveNotaryPool checks if the account is a Notary in  the SMC.
+// if it is then it checks if it has deposited , if it has it deregisters
+// itself from the pool and the lockup period counts down.
```

Comment:
Can you rephrase this? 
If I am understanding this correctly, this method has the following responsibilities: 
- Checks if account is a notary
- Checks if deposited
- Deregisters the notary
- Counts down the lockup period??

So that does four things?

Could you refactor this to just leave the notary pool? Shouldn't it be the responsibility of the caller to decide if they _should_ leave the pool? If the caller calls this method erronously, then we would expect this transaction to fail and just return the error.

In otherwords:

```
if notaryInPool() {
    leaveNotaryPool()
}
```